### DISCUSSION TITLE
How to write the file “machine.json” which can be used in  processes such as "init", “run”, "auto_test"

### DISCUSSION THREAD

**Discussion by warriors1523** [deepmodeling/dpgen#684] [Q&A]
All tasks are executed on a high performance computing platform, where the job scheduling system is slurm. Our "machine.json" file is shown in the image below. When executing the init task we can complete stage one and two. Does one "machine.json" file work for all dpgen tasks?
![image](https://user-images.githubusercontent.com/92622311/160270453-b91afd56-9590-49cc-97d5-bc1228f605db.png)
![image](https://user-images.githubusercontent.com/92622311/160270496-2925fd6e-d873-4236-98f9-f0c26acdfa8c.png)
![image](https://user-images.githubusercontent.com/92622311/160270519-9deff304-8ea6-4a82-b0f3-a04ec33b0e94.png)

**Comment by taipinghu** [comment]
One machine.json file will work for all dpgen tasks.
If your tasks are "dpgen init or dpgen auto_test" (only need CPU), the fp parameters in machine will be used.
